### PR TITLE
[DROOLS-6856] prevent overflow of segment and path memories masks

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6856

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/phreak/SegmentUtilities.java [hunk 1]
```diff
@@ -60,15 +60,12 @@
     /**
      * Initialises the NodeSegment memory for all nodes in the segment.
      */
-    public static void createSegmentMemory(LeftTupleSource tupleSource, ReteEvaluator reteEvaluator) {
```
Comment (mariofusco) [inline drools-core/src/main/java/org/drools/core/phreak/SegmentUtilities.java:63]: This change is only a minor refactor not related to the actual fix.

File: drools-core/src/main/java/org/drools/core/phreak/SegmentUtilities.java [hunk 2]
```diff
@@ -181,17 +179,25 @@ public static SegmentMemory createSegmentMemory(LeftTupleSource tupleSource, Mem
...
 
-        nodeTypesInSegment = updateRiaAndTerminalMemory(tupleSource, tupleSource, smem, reteEvaluator, false, nodeTypesInSegment);
+        updateRiaAndTerminalMemory(tupleSource, tupleSource, smem, reteEvaluator, false, nodeTypesInSegment);
 
         reteEvaluator.getKnowledgeBase().registerSegmentPrototype(segmentRoot, smem);
 
         return smem;
     }
 
+    public static long nextNodePosMask(long nodePosMask) {
```
Comment (mariofusco) [inline drools-core/src/main/java/org/drools/core/phreak/SegmentUtilities.java:189]: This is the actual fix.

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#350` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 453
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/350/testReport/org.kie.kogito.springboot.outbox/OutboxIT/___/">org.kie.kogito.springboot.outbox.OutboxIT.(?)</a></summary>
java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49237] should be listening)<br/>	at org.kie.kogito.springboot.outbox.OutboxIT.before(OutboxIT.java:83)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49237] should be listening)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/350/display/redirect

**Comment by kie-ci1** [comment]
**(tests) - kogito-apps job** `#347` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 2684
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/347/testReport/org.kie.kogito.index.mongodb.query/DomainQueryIT/test/">org.kie.kogito.index.mongodb.query.DomainQueryIT.test</a></summary>
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException<br/>Caused by: java.lang.reflect.InvocationTargetException<br/>Caused by: java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource org.kie.kogito.testcontainers.quarkus.MongoDBQuarkusTestResource@b7457b6<br/>Caused by: java.lang.RuntimeException: Unable to start Quarkus test resource org.kie.kogito.testcontainers.quarkus.MongoDBQuarkusTestResource@b7457b6<br/>Caused by: java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/347/display/redirect

**Review by tkobayas — APPROVED** [review]

**Pull request merged**